File: php/features.php
<?php
declare(strict_types=1);

// KonkanRailwayLivePosition SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class KonkanRailwayLivePositionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KonkanRailwayLivePositionBaseFeature();
            case "ratelimit":
                return new KonkanRailwayLivePositionRatelimitFeature();
            case "retry":
                return new KonkanRailwayLivePositionRetryFeature();
            case "test":
                return new KonkanRailwayLivePositionTestFeature();
            case "timeout":
                return new KonkanRailwayLivePositionTimeoutFeature();
            default:
                return new KonkanRailwayLivePositionBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: py/konkanrailwayliveposition_sdk/features.py
# KonkanRailwayLivePosition SDK feature factory

from konkanrailwayliveposition_sdk.feature.base_feature import KonkanRailwayLivePositionBaseFeature
from konkanrailwayliveposition_sdk.feature.ratelimit_feature import KonkanRailwayLivePositionRatelimitFeature
from konkanrailwayliveposition_sdk.feature.retry_feature import KonkanRailwayLivePositionRetryFeature
from konkanrailwayliveposition_sdk.feature.test_feature import KonkanRailwayLivePositionTestFeature
from konkanrailwayliveposition_sdk.feature.timeout_feature import KonkanRailwayLivePositionTimeoutFeature


_FEATURES = {
    "base": lambda: KonkanRailwayLivePositionBaseFeature(),
    "ratelimit": lambda: KonkanRailwayLivePositionRatelimitFeature(),
    "retry": lambda: KonkanRailwayLivePositionRetryFeature(),
    "test": lambda: KonkanRailwayLivePositionTestFeature(),
    "timeout": lambda: KonkanRailwayLivePositionTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES
